File: src/main/java/com/viaversion/aas/provider/AspirinVersionProvider.java
package com.viaversion.aas.provider;

import com.viaversion.aas.handler.MinecraftHandler;
import com.viaversion.viaversion.api.connection.UserConnection;
import com.viaversion.viaversion.protocols.base.BaseVersionProvider;

public class AspirinVersionProvider extends BaseVersionProvider {
	@Override
	public int getClosestServerProtocol(UserConnection connection) throws Exception {
		var ver = connection.getChannel().pipeline().get(MinecraftHandler.class).getData().getBackServerVer();
		if (ver != null) return ver;
		return super.getClosestServerProtocol(connection);
	}
}

File: src/main/kotlin/com/viaversion/aas/provider/AspirinVersionProvider.kt
package com.viaversion.aas.provider

import com.viaversion.aas.handler.MinecraftHandler
import com.viaversion.viaversion.api.connection.UserConnection
import com.viaversion.viaversion.protocols.base.BaseVersionProvider

object AspirinVersionProvider : BaseVersionProvider() {
    override fun getClosestServerProtocol(connection: UserConnection): Int {
        val ver = connection.channel!!.pipeline().get(MinecraftHandler::class.java).data.viaBackServerVer
        if (ver != null) return ver
        return super.getClosestServerProtocol(connection)
    }
}
